morphir/runtime/src/org/finos/morphir/runtime/quick/EvaluatorQuick.scala

```diff
@@ -324,7 +325,7 @@ object EvaluatorQuick {
         curry(constructor, mappedArgs)
       case (a, b)    => V.tuple(Chunk(scalaToIR(a), scalaToIR(b)))
       case (a, b, c) => V.tuple(Chunk(scalaToIR(a), scalaToIR(b), scalaToIR(c)))
-      case other     => throw new Exception(s"I don't know how to decompose $other")
+      case other     => throw new UnmatchedPattern(s"I don't know how to decompose $other")
```

Comment:
UnmatchedPattern is specifically for pattern matching in terms of morphir - this should be UnsupportedType